### PR TITLE
#4337 hover card updates

### LINKED ISSUE #4337: User hover card v2
Follow up for making for user hover card 
- add a Twitter link to the design
- copy address feature

@exezbcz will drop updated design 👀

_Originally posted by @yangwao in https://github.com/kodadot/nft-gallery/issues/4117#issuecomment-1316983217_

### PR DESCRIPTION
**Pull request by prachi00**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).

👇 \_ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring

## Context

- [x] Closes #4337
- [ ] Requires deployment <>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

#### Optional

- [ ] I've tested it at </bsx/collection>
- [ ] I've tested PR on mobile
- [x] I've written unit tests 🧪
- [ ] I've found edge cases

#### Had issue bounty label?

- [x] Fill up your KSM address: 
[Payout](https://beta.kodadot.xyz/transfer/?target=EzGc4s9PgCPx1YnF3fqzhLzVHpHMTL4LWPScwpDrR8JKgSU)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.
Screen Shot 2022-11-17 at 8 18 19 PM

### CODE REVIEW

File: components/identity/module/IdentityPopoverHeader.vue [hunk 1]
```diff
@@ -19,13 +38,19 @@ const shortenedAddress = inject('shortenedAddress')
 
 const identity = inject<{ [x: string]: string }>('identity')
 const { urlPrefix } = usePrefix()
+const { $buefy } = useNuxtApp()
+
+const toast = (message: string) => {
+  $buefy.toast.open(message)
+}
```
Comment (roiLeo) [inline components/identity/module/IdentityPopoverHeader.vue:45]: Check out new toast design from #4341 

just add:
```ts 
    message,
    type: 'is-neo',
```

File: components/identity/module/IdentityPopoverHeader.vue [hunk 2]
```diff
@@ -1,13 +1,32 @@
...
+          data-cy="identity-clipboard"
+          @click.native="toast('Copied to clipboard')"></b-icon>
+      </div>
+      <a
+        v-if="identity?.twitter"
+        :href="`https://twitter.com/${identity?.twitter}`"
+        target="_blank"
+        rel="noopener noreferrer"
+        data-cy="identity-twitter">
+        <b-icon pack="fab" icon="twitter" />
```
Comment (roiLeo) [inline components/identity/module/IdentityPopoverHeader.vue:27]:
```suggestion
        <b-icon pack="fab" type="is-link" icon="twitter" />
```

File: components/identity/module/IdentityPopoverHeader.vue [hunk 3]
```diff
@@ -1,13 +1,32 @@
...
-      class="is-size-6 mb-1 break-word"
-      :to="`/${urlPrefix}/u/${address}`">
-      <span data-cy="identity-display">
-        {{ identity?.display || shortenedAddress }}</span
-      >
-    </nuxt-link>
+    <div class="is-flex is-align-items-center is-justify-content-space-between">
+      <div class="is-flex is-align-items-center">
+        <nuxt-link
+          class="is-size-6 break-word mr-2"
```
Comment (roiLeo) [inline components/identity/module/IdentityPopoverHeader.vue:7]:
```suggestion
          class="is-size-6 break-word mr-2 has-text-link"
```

File: components/identity/module/IdentityPopoverHeader.vue [hunk 4]
```diff
@@ -1,13 +1,32 @@
...
+            {{ identity?.display || shortenedAddress }}</span
+          >
+        </nuxt-link>
+        <b-icon
+          v-clipboard:copy="address"
+          icon="copy"
+          size="is-small"
+          class="copy-icon"
+          data-cy="identity-clipboard"
+          @click.native="toast('Copied to clipboard')"></b-icon>
```
Comment (roiLeo) [inline components/identity/module/IdentityPopoverHeader.vue:19]:
```suggestion
        <b-icon
          v-clipboard:copy="address"
          icon="copy"
          size="is-small"
          type="is-link is-clickable"
          data-cy="identity-clipboard"
          @click.native="toast('Copied to clipboard')" />
```

Comment (roiLeo) [reply components/identity/module/IdentityPopoverHeader.vue:19]: add `type="is-link is-clickable"` and remove class

File: components/identity/module/IdentityPopoverHeader.vue [hunk 5]
```diff
@@ -19,13 +38,19 @@ const shortenedAddress = inject('shortenedAddress')
...
+}
 </script>
 
 <style lang="scss" scoped>
 @import '@/styles/abstracts/variables';
 
 .copy-icon {
   cursor: pointer;
+  color: $k-blue;
 }
```
Comment (roiLeo) [inline components/identity/module/IdentityPopoverHeader.vue:54]:
```suggestion
```

File: components/identity/module/IdentityPopoverHeader.vue [hunk 6]
```diff
@@ -45,6 +70,9 @@ const { urlPrefix } = usePrefix()
 
   a {
     color: $k-blue !important;
+    &:hover {
+      color: $k-blue !important;
+    }
   }
 }
```
Comment (roiLeo) [inline components/identity/module/IdentityPopoverHeader.vue:77]:
```suggestion
```

**Review by roiLeo — COMMENTED** [review]
To improve refacto see suggested change

We could add Discod link too in IdentityPopover

### Side note:
Found a bug when logged out and hovering address, Highest sales cards have green overlay (which mean I'm currentOwner but this is not the case)

Capture d’écran 2022-11-18 à 8 17 23 AM

**Comment by yangwao** [comment]
looking for new card 😏

**Comment by yangwao** [comment]
okay works 
(i have slow net in coffee place)
image

pay 50

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 24.11 USD/KSM ~ 2.074 $KSM
🧗 [EzGc4s9PgCPx1YnF3fqzhLzVHpHMTL4LWPScwpDrR8JKgSU](https://kusama.subscan.io/account/EzGc4s9PgCPx1YnF3fqzhLzVHpHMTL4LWPScwpDrR8JKgSU)
🔗 [0x2f81dd83ed9fb5c5c84a59f603f29bde0f715668c163cd015576b2345a68d789](https://kusama.subscan.io/extrinsic/0x2f81dd83ed9fb5c5c84a59f603f29bde0f715668c163cd015576b2345a68d789)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Comment by roiLeo** [comment]
sorry forgot to review this one, lgtm!

**Pull request merged**